tools/c7n_azure/c7n_azure/resources/appserviceplan.py

```diff
@@ -143,14 +144,13 @@ def _process_resource(self, resource):
         if resource['kind'] == 'linux':
             model.reserved = True
 
-        if 'size' in self.data:
-            size = Lookup.extract(self.data.get('size'), resource)
-            model.sku = models.SkuDescription()
-            model.sku.tier = ResizePlan.get_sku_name(size)
-            model.sku.name = size
+        size = Lookup.extract(self.data.get('size'), resource)
+        model.sku = models.SkuDescription()
+        model.sku.tier = ResizePlan.get_sku_name(size)
+        model.sku.name = size
 
         if 'count' in self.data:
-            model.target_worker_count = Lookup.extract(self.data.get('count'), resource)
+            model.sku.capacity = Lookup.extract(self.data.get('count'), resource)
```

Comment:
what's the difference between `capacity` and `target_worker_count`?

Reply:
These are the correct patch fields - they seem to have some dedicated patch parameters.  Maybe the bug just goes away once we use these and populating both isn't even required anymore? 

Originally I know we were doing it the same was as the CLI - but looking at the docs now we were doing it very wrong, not sure if CLI ever updated.

Reply:
So, I tried both hitting the REST API directly and using these parameters using the [AppServicePlanPatchResource](https://docs.microsoft.com/en-us/python/api/azure-mgmt-web/azure.mgmt.web.models.app_service_plan_patch_resource.appserviceplanpatchresource?view=azure-python) model and I got the following response:



```
{
  "Code": "BadRequest",
  "Message": "The parameter SKU cannot be null.",
  "Target": null,
  "Details": [
    {
      "Message": "The parameter SKU cannot be null."
    },
    {
      "Code": "BadRequest"
    },
    {
      "ErrorEntity": {
        "MessageTemplate": "The parameter {0} cannot be null.",
        "Parameters": [
          "SKU"
        ],
        "Code": "BadRequest",
        "Message": "The parameter SKU cannot be null."
      }
    }
  ],
  "Innererror": null
}
```

Unfortunately, this may have been an old bug. I am copying the Azure CLI implementation found [here](https://github.com/Azure/azure-cli/blob/f0bafa4f6122f61bfdad42b38e1c9df768eaa2ae/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1294). They set capacity property on `sku` oppose to `target_worker_count`.